llm/server.go

```diff
@@ -796,6 +796,14 @@ func (s *llmServer) Completion(ctx context.Context, req CompletionRequest, fn fu
 			if err := json.Unmarshal(evt, &c); err != nil {
 				return fmt.Errorf("error unmarshalling llm prediction response: %v", err)
 			}
+			// convert internal done reason to one of our standard api format done reasons
+			switch c.DoneReason {
+			case "limit":
```

Comment:
We might want to just return an enum from the runner instead of a string. That could help to keep things a little more focused. Even now I think we already have too many different reasons that we return - maybe just limit, stop, other?

Reply:
Switch to an enum but left the current done reason response strings for now adding a new one (other) seems outside the scope of this PR to me. Could confuse some clients.

Reply:
I agree, we shouldn't change any external behavior here.

However, I was actually trying to avoid putting a string in `seq.doneReason` because we already have a sprawl of random reasons. If we use an enum there (I was imagining that we would basically be storing an int) then that can force it to be a little more structured - we could start with `limit`, `stop`, `other`. We would still translate from that to the final outbound API string in the Ollama server.

Sorry for all of the back and forth on this.

Reply:
Should be good now, I've moved the done reason iota all the way down to the runners. It was definitely confusing with all the strings so this was a good suggestion.